enterprise/reporting/error_reporting.go

```diff
@@ -188,7 +191,7 @@ func (edr *ErrorDetailReporter) Report(ctx context.Context, metrics []*types.PUR
 			"sourceId":      metric.ConnectionDetails.SourceID,
 			"destinationId": metric.ConnectionDetails.DestinationID,
 		}).Count(int(metric.StatusDetail.Count))
-
+		sampleEvent, _ := json.Marshal(metric.StatusDetail.SampleEvent)
```

Comment:
Should we marshal in the first place ?

Reply:
If we marshal in the first place we need to change in many places as sampleEvent being used in many places.